backend/controllers/controller_kubernetes_resources.go

```diff
@@ -122,15 +120,22 @@ func getNamespace(r *http.Request) string {
 }
 
 func writeJSONResponse(w http.ResponseWriter, statusCode int, data interface{}) {
-	w.WriteHeader(statusCode)
-	json.NewEncoder(w).Encode(data)
+	if w.Header().Get("Content-Type") == "" {
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	}
+	if statusCode != http.StatusOK {
+		w.WriteHeader(statusCode)
+	}
+	if err := json.NewEncoder(w).Encode(data); err != nil {
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+	}
 }
 
 func decodeJSONBody(w http.ResponseWriter, r *http.Request, dst interface{}) bool {
 	decoder := json.NewDecoder(r.Body)
 	err := decoder.Decode(dst)
 	if err != nil {
-		writeJSONResponse(w, 400, &models.ModelError{Code: 400, Message: "Invalid request body"})
+		writeJSONResponse(w, http.StatusBadRequest, &models.ModelError{Code: http.StatusBadRequest, Message: "Invalid request body"})
```

Comment:
I don't think it should write response. When it retrurns, it could lead to writing multiple responses.

Reply:
It may be my function, but context is different.

This was how i used it. It will write error, return false and then return.
```
if !decodeJSONBody(w, r, &resource.ResourceDetails) {
		return
	}
```
	
This is how you do it:
```
func UpdateResourceController(w http.ResponseWriter, r *http.Request) {
	handleResourceOperation(w, r, models.Update, func(resourceType, namespace, resourceName string) (interface{}, *models.ModelError) {
		var resource models.ResourceDetails
		if !decodeJSONBody(w, r, &resource.ResourceDetails) {
			return nil, &models.ModelError{Code: http.StatusBadRequest, Message: "Invalid request body"}
		}
		return cluster.UpdateResource(resourceType, namespace, resourceName, resource)
	})
}


result, err := operationFunc(resourceType, namespace, resourceName)
	if err != nil {
		writeJSONResponse(w, int(err.Code), err)
		return
	}
```
After error, you return and write another error.

Reply:
![image](https://github.com/user-attachments/assets/f9ccc747-eee8-42b8-badf-7146daea39c4)
I removed the header overwrite in decodeJSONBody and left it to be overwritten only in writeJSONResponse

Reply:
![cha cha real smooth](https://github.com/user-attachments/assets/f4e349d5-1af7-4931-88b2-7f1fcd376f71)